lib/tasks/packaging.rb

```diff
@@ -37,17 +37,21 @@ def deb_build_cmd(packaging_dir, work_dir, output_dir, dist, version, extra_args
   desc "Tar and zip OnDemand into packaging dir"
   task :tar, [:dist] do |task, args|
     dist = args[:dist] || 'el8'
+    cmd = ['git', 'ls-files', '|', tar, '-c']
     if dist =~ /^el/
       version = ood_package_version
       tar_file = "packaging/rpm/v#{version}.tar.gz"
     else
       dir = File.join(Dir.pwd, 'build').tap { |p| sh "mkdir -p #{p}" }
       version = ood_package_version.gsub('-', '.')
```

Comment:
WRT refactoring versions - it's kinda shown here in `version = ood_package_version.gsub('-', '.')` I just kinda don't think this knowledge belongs here.

Maybe we need some sort of raw `version` with `rpm_version` and `deb_version` that are derived from it?  In any case - it's about pushing this knowledge about what's valid for what packaging system etc somewhere more appropriate.

Reply:
Added some better logic to handle versions and also same idea for nightly versions too.